.github/workflows/publish-to-test-pypi.yml

```diff
@@ -0,0 +1,90 @@
+name: Publish Python 🐍 distribution 📦 to PyPI
+
+on: push
+
+jobs:
+  build:
+    name: Build distribution 📦
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v4
+    - name: Set up Python
+      uses: actions/setup-python@v5
+      with:
+        python-version: "3.10.12"
+    - name: Install specific setuptools and pip
+      run: |
+        python -m pip install setuptools==65.5.0 pip==21
+    - name: Install pypa/build
+      run: |
+        python3 -m pip install build --user
+    - name: Build a binary wheel and a source tarball
+      run: python3 -m build
+    - name: Store the distribution packages
+      uses: actions/upload-artifact@v4
+      with:
+        name: python-package-distributions
+        path: dist/
+
+  publish-to-pypi:
+    name: >-
+      Publish Python 🐍 distribution 📦 to PyPI
+    if: startsWith(github.ref, 'refs/tags/')  # only publish to PyPI on tag pushes
+    needs:
+    - build
+    runs-on: ubuntu-latest
+    environment:
+      name: pypi
+      url: https://pypi.org/p/nett-benchmarkse  
+    permissions:
+      id-token: write  
+
+    steps:
+    - name: Download all the dists
+      uses: actions/download-artifact@v4
+      with:
+        name: python-package-distributions
+        path: dist/
+    - name: Publish distribution 📦 to PyPI
+      uses: pypa/gh-action-pypi-publish@release/v1
+
+  github-release:
+    name: >-
+      Sign the Python 🐍 distribution 📦 with Sigstore
+      and upload them to GitHub Release
+    needs:
+    - publish-to-pypi
+    runs-on: ubuntu-latest
+
+    permissions:
+      contents: write  # IMPORTANT: mandatory for making GitHub Releases
+      id-token: write  # IMPORTANT: mandatory for sigstore
+
+    steps:
+    - name: Download all the dists
+      uses: actions/download-artifact@v4
+      with:
+        name: python-package-distributions
+        path: dist/
+    - name: Sign the dists with Sigstore
+      uses: sigstore/gh-action-sigstore-python@v2.1.1
```

Comment:
Clarification: Does this push them to pypi? I believe we need a secret added to the repo for it to publish to pypi, right? I didn't see one listed in the repo settings. Perhaps, I am just unfamiliar with how this works.

Reply:
Yes, it does push it to pypi. As for the repo setting, the docs aren't very clear on if that is required. The only part that is required is authorizing GitHub to push to pypi (just like the maintainers of the library have the authorization to do so)

<img width="761" alt="Screenshot 2024-10-13 at 7 19 56 PM" src="https://github.com/user-attachments/assets/437c0ef2-5971-4654-9952-f245933cce64">